Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.Fluent.cs

```diff
@@ -0,0 +1,209 @@
+        /// <seealso cref="IndexingPolicy.ExcludedPaths"/>
+        [IgnoreForUnitTest]
+        public CosmosContainerSettings WithExcludeIndexPath(string path)
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            ExcludedPath excludedPath = new ExcludedPath()
+            {
+                Path = path
+            };
+
+            this.IndexingPolicy.ExcludedPaths.Add(excludedPath);
+            return this;
+        }
+
+        /// <summary>
+        /// Includes the given additional composite index
+        /// </summary>
+        /// <seealso cref="IndexingPolicy.CompositeIndexes"/>
+        /// <seealso cref="CompositePathDefinition"/>
+        [IgnoreForUnitTest]
+        public CosmosContainerSettings WithCompositeIndex(params string[] compositeIndexPaths)
+        {
+            if (compositeIndexPaths == null)
+            {
+                throw new ArgumentNullException(nameof(compositeIndexPaths));
+            }
+
+            if (compositeIndexPaths.Length == 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(compositeIndexPaths));
+            }
+
+            CompositePathDefinition[] compositeSpecs = new CompositePathDefinition[compositeIndexPaths.Length];
+            for (int i = 0; i < compositeIndexPaths.Length; i++)
+            {
+                if (string.IsNullOrEmpty(compositeIndexPaths[i]))
+                {
+                    throw new ArgumentOutOfRangeException($"{nameof(compositeIndexPaths)} has null or empty entry at position {i}");
+                }
+
+                compositeSpecs[i] = new CompositePathDefinition()
+                {
+                    Path = compositeIndexPaths[i],
+                    Order = CompositePathSortOrder.Ascending,
+                };
+            }
+
+            return this.WithCompositeIndex(compositeSpecs);
+        }
+
+        /// <summary>
+        /// Includes the given additional composite index
+        /// </summary>
+        /// <seealso cref="IndexingPolicy.CompositeIndexes"/>
+        /// <seealso cref="CompositePathDefinition"/>
+        [IgnoreForUnitTest]
+        public CosmosContainerSettings WithCompositeIndex(params CompositePathDefinition[] compositeIndexPaths)
+        {
+            if (compositeIndexPaths == null)
+            {
+                throw new ArgumentNullException(nameof(compositeIndexPaths));
+            }
+
+            if (compositeIndexPaths.Length == 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(compositeIndexPaths));
+            }
+
+            Collection<CompositePathDefinition> compositePathsCollection = new Collection<CompositePathDefinition>();
+            for (int i=0; i < compositeIndexPaths.Length; i++)
+            {
+                if (compositeIndexPaths[i] == null || String.IsNullOrEmpty(compositeIndexPaths[i].Path))
```

Comment:
Could simplify with:

```suggestion
                if (string.IsNullOrEmpty(compositeIndexPaths[i]?.Path))
```